tools/run_tests/performance/loadtest_config.py

```diff
@@ -118,17 +118,77 @@ def gen_run_indices(runs_per_test: int) -> Iterable[str]:
         yield index_fmt.format(i)
 
 
+def scenario_name(base_name, client_channels, server_threads, offered_load):
+    """Constructs scenario name from base name and modifiers."""
+
+    elements = [base_name]
+    if client_channels:
+        elements.append('{:d}channels'.format(client_channels))
+    if server_threads:
+        elements.append('{:d}threads'.format(server_threads))
+    if offered_load:
+        elements.append('{:d}offered_load'.format(offered_load))
+    return '_'.join(elements)
+
+
+def scenario_transform_function(
+    client_channels: int, server_threads: int, offered_loads: Iterable[int]
```

Comment:
It looks like `None` is a permissible value for `client_channels` and `server_channels`. Their type should be `Optional[int]`.